iroh-base/src/key.rs

```diff
@@ -134,14 +134,22 @@ impl PublicKey {
 
     /// Convert to a hex string limited to the first 5 bytes for a friendly string
     /// representation of the key.
-    pub fn fmt_short(&self) -> String {
-        data_encoding::HEXLOWER.encode(&self.as_bytes()[..5])
+    pub fn fmt_short(&self) -> impl Display + 'static {
+        PublicKeyShort(*self)
     }
 
     /// The length of an ed25519 `PublicKey`, in bytes.
     pub const LENGTH: usize = ed25519_dalek::PUBLIC_KEY_LENGTH;
 }
 
+struct PublicKeyShort(PublicKey);
```

Comment:
The old code used space for 10 bytes on the heap.  This version uses 32 bytes on the stack?

I think it's a great idea to try and get rid of the heap allocation.  But maybe you can give `PublicKeyShort` only the 5 bytes needed for the display?

Reply:
Yes, you could do that easily. PublicKeyShort is not part of the public API, so you can optimize it.